packages/rocketchat-livechat/server/lib/Livechat.js

```diff
@@ -60,13 +65,15 @@ RocketChat.Livechat = {
 
 			RocketChat.models.Rooms.insert(room);
 			RocketChat.models.Subscriptions.insert(subscriptionData);
+
+			newRoom = true;
 		} else {
 			room = Meteor.call('canAccessRoom', message.rid, guest._id);
 		}
 		if (!room) {
 			throw new Meteor.Error('cannot-acess-room');
 		}
-		return RocketChat.sendMessage(guest, message, room);
+		return _.extend(RocketChat.sendMessage(guest, message, room), { newRoom: newRoom });
 	},
 	registerGuest({ token, name, email, department, phone, loginToken, username } = {}) {
```

Comment:
![Codacy](https://www.codacy.com/assets/images/favicon.png) Issue found: [This function's cyclomatic complexity is too high. (12)](https://www.codacy.com/app/RocketChat/Rocket-Chat/file/2502285622/issues/source?bid=3199634&fileBranchId=3273411#l78)